fe/fe-core/src/main/java/org/apache/doris/datasource/property/fileformat/JsonFileFormatProperties.java

```diff
@@ -0,0 +1,122 @@
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.datasource.property.fileformat;
+
+import org.apache.doris.common.util.FileFormatConstants;
+import org.apache.doris.common.util.Util;
+import org.apache.doris.nereids.exceptions.AnalysisException;
+import org.apache.doris.proto.InternalService.PFetchTableSchemaRequest;
+import org.apache.doris.thrift.TFileAttributes;
+import org.apache.doris.thrift.TFileFormatType;
+import org.apache.doris.thrift.TFileTextScanRangeParams;
+import org.apache.doris.thrift.TResultFileSinkOptions;
+
+import java.util.Map;
+
+public class JsonFileFormatProperties extends FileFormatProperties {
+    // from ExternalFileTableValuedFunction:
+    private String jsonRoot = "";
+    private String jsonPaths = "";
+    private boolean stripOuterArray;
+    private boolean readJsonByLine;
+    private boolean numAsString;
+    private boolean fuzzyParse;
+
+
+    public JsonFileFormatProperties(TFileFormatType fileFormatType) {
+        super(fileFormatType);
+    }
+
+    @Override
+    public void analyzeFileFormatProperties(Map<String, String> formatProperties, boolean isRemoveOriginProperty)
+            throws AnalysisException {
+        // 这几个json应该移到json checker中
+        jsonRoot = getOrDefaultAndRemove(formatProperties, FileFormatConstants.PROP_JSON_ROOT,
+                "", isRemoveOriginProperty);
+        jsonPaths = getOrDefaultAndRemove(formatProperties, FileFormatConstants.PROP_JSON_PATHS,
+                "", isRemoveOriginProperty);
+        readJsonByLine = Boolean.valueOf(
+                getOrDefaultAndRemove(formatProperties, FileFormatConstants.PROP_READ_JSON_BY_LINE,
+                        "", isRemoveOriginProperty)).booleanValue();
+        stripOuterArray = Boolean.valueOf(
+                getOrDefaultAndRemove(formatProperties, FileFormatConstants.PROP_STRIP_OUTER_ARRAY,
+                        "", isRemoveOriginProperty)).booleanValue();
+        numAsString = Boolean.valueOf(
+                getOrDefaultAndRemove(formatProperties, FileFormatConstants.PROP_NUM_AS_STRING,
+                        "", isRemoveOriginProperty)).booleanValue();
+        fuzzyParse = Boolean.valueOf(
+                getOrDefaultAndRemove(formatProperties, FileFormatConstants.PROP_FUZZY_PARSE,
+                        "", isRemoveOriginProperty)).booleanValue();
+
+        String compressTypeStr = getOrDefaultAndRemove(formatProperties, FileFormatConstants.PROP_COMPRESS_TYPE,
+                "UNKNOWN", isRemoveOriginProperty);
+        try {
+            compressionType = Util.getFileCompressType(compressTypeStr);
+        } catch (IllegalArgumentException e) {
+            throw new AnalysisException("Compress type : " +  compressTypeStr + " is not supported.");
```

Comment:
I think we should let `Util.getFileCompressType` to handle the exception.
If exception occurred, just return `TFileCompressType.UNKNOWN`

Reply:
done